packages/components/src/dropdown-menu/index.js

```diff
@@ -30,6 +30,16 @@ function mergeProps( defaultProps = {}, props = {} ) {
 	return mergedProps;
 }
 
+function callAll( ...fns ) {
```

Comment:
Is it really necessary? It seems like it would be easier to follow if those functions calls would be inlined like:

```js
onClick={
    ( ...args ) => {
        onToggle( ...args );
        mergedToggleProps. onClick( ...args );
    }
}
```

Even better, we could consider having this logic embedded into `mergeProps` helper. Feel free to update it if this would simplify things further.